selinux/bluechi.fc

```diff
@@ -4,3 +4,5 @@
 /usr/bin/bluechi-agent	--	gen_context(system_u:object_r:bluechi_agent_exec_t,s0)
 /usr/libexec/bluechi-agent	--	gen_context(system_u:object_r:bluechi_agent_exec_t,s0)
 
+/run/bluechi(/.*)?		gen_context(system_u:object_r:bluechi_var_run_t,s0)
+/var/run/bluechi(/.*)?		gen_context(system_u:object_r:bluechi_var_run_t,s0)
```

Comment:
This is not right. /var/run is typically a symlink to /run, and there is an alias in the selinux policy that resolves this:
https://github.com/fedora-selinux/selinux-policy/blob/66b9a6fd5b355b5b0ff782098addd4398ec07303/config/file_contexts.subs_dist#L14

So, you only need to use one of these, the "canonical" representation of /run. I believe for RHEL9 this is /var/run, but this was switched around in f40 (https://discussion.fedoraproject.org/t/f40-change-proposal-move-var-run-selinux-policy-entries-to-run-self-contained/100171/4 ) to /run.

Reply:
> This is not right. /var/run is typically a symlink to /run, and there is an alias in the selinux policy that resolves this:
https://github.com/fedora-selinux/selinux-policy/blob/66b9a6fd5b355b5b0ff782098addd4398ec07303/config/file_contexts.subs_dist#L14

Ah, didn't know about these aliases. I used other applications like dbus and systemd as a template. If we remove `/var/run` the probably the type should also be `bluechi_run_t` instead of `bluechi_var_run_t`, right?

Reply:
Yes in Fedora now it is /run and in RHEL < 10 it is /var/run.

Reply:
Yes, /var/run is canonical in rhel9, /run in rhel10

Reply:
So, lets keep both which hopefully makes this work with both 9 and 10. However, maybe a comment about this?

Reply:
Added a comment above the file context lines.